.vscode/settings.json

```diff
@@ -0,0 +1,11 @@
+{
```

Comment:
I've never used vscode before, is this the right place to put `"files.trimTrailingWhitespace": true`?

Reply:
Yes, and I've just added that below. 

I feel conflicted about that, because my initial intent with this PR was for the `extensions.json` file above to encourage people to just add the [editorconfig](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig) plugin, and then have all whitespace config settings in the `.editorconfig` file [here](https://github.com/ethereum-optimism/optimism/blob/develop/.editorconfig#L10).

But if adding this one line fixes the biggest whitespace pain point instantly, it seems reasonable to just include.

Reply:
I'm not super opinionated and never really have messed with `.vscode` or `.editorconfig` files